tutorials/python/python-type-driven-development/tutorial.md

```diff
@@ -0,0 +1,207 @@
+---
+title: Type Driven Development in Python
+published: 2019-08-01
+tags: Python, typing, types, tdd
+language: python
+author-name: Carlos Villavicencio
+twitter-profile: po5i
+github-profile: po5i
+description: Learn how to improve the software quality and readability by using type driven development in Python.
+---
+
+_Type Driven Development_ can increase the performance, accuracy, and readability of your code by taking advantage of the language’s type system. There are many books that you can use to learn how to achieve this, for example, “Type-Driven Development with Idris” ([Manning][tdd-idris]) and extrapolate these practices to other languages.
+
+In a [previous article][types-vs-tests], we discussed the benefits of working with _Type Driven Development_ and the importance of static type checking in compilation time for strongly typed languages such as Java or Haskell. It doesn’t matter if the paradigm is object-oriented or functional, type-checking adds a layer of safety, reducing the amount of tests we need to write, and reduces our dependency on tests.
+
+## What about Python?
+
+Is it possible to use a strong type system in a language like Python? The answer is yes. [PEP-484][pep-484] introduced _Type Annotations to Python_ 3.5 back in 2014. It’s been a while, right? Why this isn’t popular? Because Python doesn’t require the code has types everywhere to run, they’re completely optional and mostly used by editors and IDEs. Another reason is that Python code is well-known to be readable in the least number of possible lines of code, but adding types to your code doesn’t make it more readable? What a paradox!
+
+In this article, we’ll be walking through the Python type system, how to add them to our code and Docstrings and perform static type checking using [mypy][mypy]. There are plenty options for static type-checking in Python, but we'll follow [Guido van Rossum suggestion][guido-mypy].
+
+### Mypy
+
+Mypy is a third-party Python library that provides an optional static type check. Opposite of other non-dynamic programming languages like Java, where the static type-check takes place at compilation time, Mypy CLI does the type-check to a file (or a set of files), so it's up to you to decide where in your workflow the type-checking will be executed, it can be on CI or a pre-commit hook, just to mention some.
```

Comment:
One more taking into account that this is a tutorial. Maybe it would help if there's an example (it could be a separate repository) with a small Python project that has this configured in CI or in whatever way you'd recommend.

Reply:
In the previous paragraph it's explaned why we're choosing `mypy` instead of any other option available. I also rephrased the Pip-related line.

> One more taking into account that this is a tutorial. Maybe it would help if there's an example (it could be a separate repository) with a small Python project that has this configured in CI or in whatever way you'd recommend.

Good idea, I will think about expanding the examples to a bigger one so I can cover TypeDD better, like you suggested in the last comment of your review.